tools/serve/test_serve.py

```diff
@@ -107,3 +107,41 @@ def test_alternate_host_invalid(primary, alternate):
 ])
 def test_alternate_host_valid(primary, alternate):
     ConfigBuilder(logger, browser_host=primary, alternate_hosts={"alt": alternate})
+
+
+# A token marking the location of expected script injection.
+INJECT_SCRIPT_MARKER = b"<!-- inject here -->"
+
+
+def test_inject_script_after_head():
+    html = b"""<!DOCTYPE html>
+    <html>
+        <head>
+        <!-- inject here --><script src="test.js"></script>
+        </head>
+        <body>
+        </body>
+    </html>"""
+    assert INJECT_SCRIPT_MARKER in html
+    assert inject_script(html.replace(INJECT_SCRIPT_MARKER, b""), INJECT_SCRIPT_MARKER) == html
```

Comment:
Yay for these tests, and also a nice way here of not writing out both expected and actual as literals.